go/consensus/tendermint/apps/staking/state/gas.go

```diff
@@ -46,14 +46,17 @@ func AuthenticateAndPayFees(
 		return nil
 	}
 
+	// Convert signer's public key to account address.
+	addr := staking.NewAddress(signer)
```

Comment:
Does something prohibit `addr` from being a reserved one?

Reply:
Agreed. Added the check.